src/lib.rs

```diff
@@ -358,6 +438,24 @@ fn effective_value(output: &OutputGroup, feerate: f32) -> u64 {
         .saturating_sub(calculate_fee(output.weight, feerate))
 }
 
+fn generate_random_bool() -> bool {
+    // Create a random number generator
+    let mut rng = rand::thread_rng();
+    
+    // Generate a random boolean value
+    rng.gen()
+}
+pub fn desce_input<'a>(inputs: &'a [OutputGroup]) -> Vec<(usize, OutputGroup)> {
```

Comment:
Cloning of the inputs are also not needed.
Refer srd and fifo implementation.
Vec<(usize, &OutputGroup)> is all you need.

This function also is not likely to be used by other top level functions.
So remove this function.